tests/testthat/test-install-torsten.R

```diff
@@ -0,0 +1,157 @@
+## Adapted from test-install.R from the cmdstanr package
+
+context("install_torsten")
+
+torsten_test_tarball_url <- Sys.getenv("TORSTEN_TEST_TARBALL_URL")
+if (!nzchar(torsten_test_tarball_url)) {
+  torsten_test_tarball_url <- NULL
+}
+
+test_that("install_torsten() successfully installs torsten", {
+  if (getRversion() < '3.5.0') {
+    dir <- tempdir()
+  } else {
+    dir <- tempdir(check = TRUE)
+  }
+  expect_message(
+    expect_output(
+      install_torsten(dir = dir, cores = 2, quiet = FALSE,
+                      release_url = torsten_test_tarball_url),
+      "--- Torsten",
+      fixed = TRUE
+    ),
+    "* Finished installing Torsten",
+    fixed = TRUE
+  )
+})
+
+# test_that("install_torsten() errors if it times out", {
+#   if (getRversion() < '3.5.0') {
+#     dir <- tempdir()
+#   } else {
+#     dir <- tempdir(check = TRUE)
+#   }
+#   # with quiet=TRUE
+#   expect_warning(
+#     expect_message(
+#       install_torsten(dir = dir, timeout = 1, quiet = TRUE,
+#                       release_url = torsten_test_tarball_url),
+#       "* * Installing Torsten from https://github.com",
+#       fixed = TRUE
+#     ),
+#     "increasing the value of the 'timeout' argument and running again with 'quiet=FALSE'",
+#     fixed = TRUE
+#   )
+#   # with quiet=FALSE
+#   expect_warning(
+#     expect_message(
+#       install_cmdstan(dir = dir, timeout = 1, quiet = FALSE,
+#                       release_url = cmdstan_test_tarball_url),
+#       "* * Installing Torsten from https://github.com",
+#       fixed = TRUE
+#     ),
+#     "Try increasing the value of the 'timeout' argument.",
+#     fixed = TRUE
+#   )
+# })
+
+test_that("install_torsten() errors if invalid version or URL", {
+  expect_error(
+    install_torsten(version = "0.89.2"),
+    "Available Torsten versions do not include 0.89.2"
+  )
+  expect_error(
+    install_torsten(release_url = "https://github.com/metrumresearchgroup/Torsten/archive/refs/tags/torsten_v0.89.2.tar.gz"),
+    "Download of Torsten failed. Please check if the supplied release URL is valid."
+  )
+  expect_error(
+    install_torsten(release_url = "https://github.com/metrumresearchgroup/Torsten/archive/refs/tags/0.89.2"),
+    "cmdstanr supports installing from .tar.gz archives only"
+  )
+})
+
+test_that("install_torsten() works with version and release_url", {
+  if (getRversion() < '3.5.0') {
+    dir <- tempdir()
+  } else {
+    dir <- tempdir(check = TRUE)
+  }
+
+  expect_message(
+    expect_output(
+      install_torsten(dir = dir, cores = 4,
```

Comment:
This and other spots should avoid assuming the test machine has a given number of `cores`. It can use the default, and the tester can adjust the `mc.cores` option as desired.

Reply:
How about the alternative of using `parallel::detectCores`, e.g., `install_torsten(dir = dir, cores = min(parallel::detectCores(), 4)` or `install_torsten(dir = dir, cores = min(parallel::detectCores(), getOption("mc.cores"))`? That way the test can take advantage of available cores even if the user has not set the `mc.cores` option.

Reply:
Hmm, I do see the appeal of letting developers get a speedup without manually configuring `mc.cores`.  While I don't think `mc.cores` is too surprising of a knob to tweak given the `install_cmdstan` and `install_torsten` default for cores is `getOption("mc.cores", 2)`, developers likely aren't going to go looking for it at all.

However, a complication here is choosing a method that keeps the tests well behaved in various scenarios, such as running `R CMD check` with `_R_CHECK_LIMIT_CORES_=TRUE` (as `devtools::test()` does by default), running in a container with a CPU quota, or running on a grid worker with the available CPUs set by scheduler (as is the case for test run by MPN build process).

Your suggestions would probably work well for most manual testing, aside from the edge case of `detectCores` returning `NA`.  Even for `cores = min(parallel::detectCores(), 4)` on machine with 4 cores or fewer, we're probably not worried about competing with too much else that's running on a manual tester's machine.

On the other hand, those suggestions wouldn't play nicely with some of the testing scenarios I mentioned above.  For example, under `_R_CHECK_LIMIT_CORES_=TRUE`, the tests would fail.

So, perhaps we can optionally use [parallelly::availableCores](https://parallelly.futureverse.org/reference/availableCoresa.html), which accounts for these things.  We could:

 * add `parallelly` the set of suggested packages in `DESCRIPTION`
 * add a test helper that uses `availableCores` if it's installed

The test helper could be something like this (untested):

```R
available_cores <- function() {
  if (!requireNamespace("parallelly", quietly = TRUE)) {
    return(getOption("mc.cores", 2))
  }
  return(unname(parallelly::availableCores(omit = 1)))
}
```